File: src/main/java/sk/hudak/prco/manager/impl/UpdateProductDataListenerAdapter.java
package sk.hudak.prco.manager.impl;

import lombok.extern.slf4j.Slf4j;
import sk.hudak.prco.manager.UpdateProductDataListener;
import sk.hudak.prco.manager.UpdateStatusInfo;

@Slf4j
public class UpdateProductDataListenerAdapter implements UpdateProductDataListener {

    public static UpdateProductDataListener INSTANCE = new UpdateProductDataListenerAdapter();

    public static UpdateProductDataListener LOG_INSTANCE = updateStatusInfo -> log.debug(updateStatusInfo.toString());

    @Override
    public void onUpdateStatus(UpdateStatusInfo updateStatusInfo) {
        // do nothing
    }
}

File: src/main/java/sk/hudak/prco/manager/updateprocess/UpdateProductDataListenerAdapter.kt
package sk.hudak.prco.manager.updateprocess

import org.slf4j.LoggerFactory

class UpdateProductDataListenerAdapter : UpdateProductDataListener {

    companion object {
        val log = LoggerFactory.getLogger(UpdateProductDataListenerAdapter::class.java)!!

        val EMPTY_INSTANCE: UpdateProductDataListener = UpdateProductDataListenerAdapter()

        val LOG_INSTANCE: UpdateProductDataListener = object : UpdateProductDataListener {
            override fun onUpdateStatus(updateStatusInfo: UpdateStatusInfo) {
                log.debug(updateStatusInfo.toString())
            }
        }
    }

    override fun onUpdateStatus(updateStatusInfo: UpdateStatusInfo) {
        // do nothing
    }
}
